resource.go

```diff
@@ -141,11 +142,11 @@ func ensureDefaultPatches(clusterStatus providerv1alpha1.StatusCluster, patches
 	nodesEmpty := clusterStatus.Nodes == nil
 	versionsEmpty := clusterStatus.Versions == nil
 
-	if conditionsEmpty && nodesEmpty && versionsEmpty {
+	if reflect.DeepEqual(clusterStatus, providerv1alpha1.StatusCluster{}) {
 		patches = append(patches, Patch{
 			Op:    "add",
-			Path:  "/status",
-			Value: Status{},
+			Path:  "/status/cluster",
+			Value: providerv1alpha1.StatusCluster{},
```

Comment:
I don't really understand how this solves the issue you described - did you test that this works?

Why would the status have been reset in the first place?

Reply:
These patches are not typical UpdateStatus requests. Adding Status{} on /status path was also deleting already set /status/cluster/network - we saw in aws-operator logs after statusresource would patch status that network allocation would be triggered again for same cluster awsconfig CR for which network was already allocated.

Initial variant of the fix that was tested successfully was complete removal of the `if` block that's reinitializing `/status` - to test I was changing just the vendored statusresource code.

Tests are running now for https://github.com/giantswarm/aws-operator/pull/2090 using this branch/variant of the statusresource fix.

Reply:
Okay got it.